hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/opt/physical/CreateDagVisitor.java

```diff
@@ -139,6 +149,62 @@ public Vertex onFullScan(FullScanPhysicalRel rel) {
                 .fullScanReader(dag, table, rel.filter(parameterMetadata), rel.projection(parameterMetadata));
     }
 
+    public Vertex onMapIndexScan(IMapIndexScanPhysicalRel rel) {
+        Table table0 = rel.getTable().unwrap(HazelcastTable.class).getTarget();
+        final MapTableIndex tableIndex = rel.getIndex();
+        final PartitionedMapTable table = (PartitionedMapTable) table0;
+        final ComparatorEx<Object[]> comparator = ExpressionUtil.comparisonFn(rel.getCollations());
+
+        String indexName = tableIndex.getName();
+        IndexFilter filter = rel.getIndexFilter();
+        Expression<Boolean> remFilter = convertFilter(
+                new PlanNodeSchema(table.fieldTypes()),
+                rel.getRemainderExp(),
+                parameterMetadata
+        );
+
+        collectObjectKeys(table);
```

Comment:
Would be nice to create plan cache test(s) with indexed `IMap`.

Reply:
Done!